apps/docs/docs/learn/05-theming/04-variable-types.mdx

```diff
@@ -0,0 +1,193 @@
+as `stylex.types.color` or `stylex.types.length`.
+
+The complete list of type functions is:
+
+- `stylex.types.angle`
+- `stylex.types.color`
+- `stylex.types.url`
+- `stylex.types.image`
+- `stylex.types.integer`
+- `stylex.types.lengthPercentage`
+- `stylex.types.length`
+- `stylex.types.percentage`
+- `stylex.types.number`
+- `stylex.types.resolution`
+- `stylex.types.time`
+- `stylex.types.transformFunction`
+- `stylex.types.transformList`
+
+## Usage
+
+To assign types to variables, the value of the variable can be wrapped in
+with the appropriate type function.
+
+For example, consider the following set of variables:
+
+```tsx title="tokens.stylex.js"
+import * as stylex from '@stylexjs/stylex';
+
+export const tokens = stylex.defineVars({
+  primaryText: 'black',
+  secondaryText: '#333',
+  borderRadius: '4px',
+  angle: '0deg',
+  int: '2',
+});
+```
+
+Currently, all the values can be arbitrary strings. To assign types to the
+variables, they can be wrapped with the appropriate type function:
+
+```tsx title="tokens.stylex.js"
+import * as stylex from '@stylexjs/stylex';
+
+export const tokens = stylex.defineVars({
+  primaryText: stylex.types.color('black'),
+  secondaryText: stylex.types.color('#333'),
+  borderRadius: stylex.types.length('4px'),
+  angle: stylex.types.angle('0deg'),
+  int: stylex.types.integer(2),
+});
+```
+
+### Conditional Values
+
+The usage remains unchanged even when at-rules are used within the values.
+The following is completely valid:
+
+```tsx title="tokens.stylex.js"
+import * as stylex from '@stylexjs/stylex';
+
+export const colors = stylex.defineVars({
+  primaryText: stylex.types.color({default: 'black', [DARK]: 'white'}),
+});
+```
+
+## Type-safety in your source code
+
+The primary utility of `stylex.types.*` functions is to enable functionality
+by declaring types for variables in the generated CSS. However, the StyleX API
+also enhances the type-safety within your own codebase.
+
+When a variable is declared with a certain type within `stylex.defineVars`, the
+static types will enforce that the same type function is used when the variable
+is themed within a `stylex.createTheme` call.
+
+
+```tsx title="theme.js"
+import * as stylex from '@stylexjs/stylex';
+import {tokens} from './tokens.stylex.js';
+
+export const highContrast = stylex.defineVars({
+  primaryText: stylex.types.color('black'),
+  secondaryText: stylex.types.color('#222'),
+  borderRadius: stylex.types.length('8px'),
+  angle: stylex.types.angle('0deg'),
+  int: stylex.types.integer(4),
+});
+```
+
+Since the types for the variables are already declared within the `stylex.defineVars`
+call, the usage type functions within `stylex.createTheme` is functionally a no-op, but
+is required by the static types for type-safety.
+
+## Example use-cases
+
+### Simulating `round()`
+
+Modern browsers are [starting to support](https://caniuse.com/?search=round())
+the `round()` function in CSS. However, the feature can be simulated with
```

Comment:
Why not attach the link to 'round' instead?

If this is something that can be polyfilled, why don't we do that in stylex at some point?